src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -29,34 +29,33 @@ public static unsafe void Fill<T>(ref T refData, nuint numElements, T value)
             {
                 // We have enough data for at least one vectorized write.
 
-                T tmp = value; // Avoid taking address of the "value" argument. It would regress performance of the loops below.
                 Vector<byte> vector;
 
                 if (sizeof(T) == 1)
                 {
-                    vector = new Vector<byte>(Unsafe.As<T, byte>(ref tmp));
+                    vector = new Vector<byte>(Unsafe.BitCast<T, byte>(value));
```

Comment:
This and a similar suggestion for most other paths:
```suggestion
                    vector = Vector.Create<T>(value).As<T, byte>();
```

Reply:
The paths could likely just all be collapsed to a single `if (...)` block as well if using the above pattern, notably